src/components/Proposals/Proposal/ProposalVotingActions.tsx

```diff
@@ -26,6 +27,19 @@ export default function ProposalVotingActions({
   const openDialog = useOpenDialog();
   const { signIn } = useNear();
   const [selectedVote, setSelectedVote] = useState<number>();
+
+  const { voteIndex: userVoteIndex } = useUserVote(proposal.id);
```

Comment:
Right now we're not handling any loading states which means the UI might flicker once we fetch the user's vote.

A couple options are:
1. Show a loading fallback/skeleton for this component
2. Fetch this higher up in the tree and block the page load until this is returned